provider/kubernetes/kubernetes.go

```diff
@@ -421,9 +435,45 @@ func getRuleForPath(pa extensionsv1beta1.HTTPIngressPath, i *extensionsv1beta1.I
 		}
 		rules = append(rules, ruleTypeReplacePath+":"+rootPath)
 	}
+
+	if requestModifier := getStringValue(i.Annotations, annotationKubernetesRequestModifier, ""); requestModifier != "" {
+		rule, err := parseRequestModifier(requestModifier, ruleType)
+		if err != nil {
+			return "", err
+		}
+
+		rules = append(rules, rule)
+	}
 	return strings.Join(rules, ";"), nil
 }
 
+func parseRequestModifier(requestModifier, ruleType string) (string, error) {
+	if strings.Trim(requestModifier, " :") == "" {
+		return "", fmt.Errorf("could not get rule due to incomplete rule: %q", requestModifier)
+	}
+
+	// Split annotation to determine modifier type
+	modifierParts := strings.Split(strings.Trim(requestModifier, " :"), ":")
+	if len(modifierParts) < 2 {
+		return "", fmt.Errorf("could not get rule due to incomplete rule: %q", requestModifier)
```

Comment:
Can we make this error message different from the one in line 452 to allow distinguishing?